packages/workflow/src/internals.ts

```diff
@@ -651,8 +676,16 @@ export class Activator implements ActivationHandler {
         this.rejectUpdate(protocolInstanceId, error);
         return;
       }
+
+      const entry = this.updateHandlers.get(name);
```

Comment:
We shouldn't need this if statement; we just checked that `updateHandlers` indeed contains key `name`, and null values should not be allowed in that map.

In fact, for coherency, we should extract `entry` above, and use `validator`/`handler` from that `entry` rather than looking into `updateHandlers` every single time; i.e. once we start handling a specific incoming update request, everything relating to that update request should be handled using the same "update handler entry".

Reply:
Thanks! Tightened all that up in recent commits.